src/rgw/rgw_rest_swift.cc

```diff
@@ -572,7 +572,10 @@ int RGWCopyObj_ObjStore_SWIFT::get_params()
   dest_bucket_name = s->bucket_name_str;
   dest_object = s->object.name;
 
-  if (!s->info.x_meta_map.empty()) {
+  const char * const fresh_meta = s->info.env->get("HTTP_X_FRESH_METADATA");
+  if (fresh_meta && strcasecmp(fresh_meta, "TRUE") == 0) {
```

Comment:
With _ATTRSMOD_REPLACE_ the suppot for _X-Fresh-Metadata_ comes almost for for free.